src/core/or/relay.c

```diff
@@ -2610,7 +2610,7 @@ update_circuit_on_cmux_(circuit_t *circ, cell_direction_t direction,
              file, lineno);
     return;
   }
-  tor_assert(circuitmux_attached_circuit_direction(cmux, circ) == direction);
+  tor_assert(circuitmux_attached_circuit_direction(hashent) == direction);
```

Comment:
One alternative might be just to remove this assertion entirely -- I don't know if that's the right choice, though.